src/lib/Core.jsx

```diff
@@ -313,48 +366,132 @@ function Core({
       {renderQuizResultQuestions()}
     </div>
   );
+
+  useEffect(() => {
+    let countdown;
+
+    if (timer && isRunning && timeRemaining > 0) {
+      countdown = setInterval(() => {
+        setTimeRemaining((prevTime) => prevTime - 1);
+      }, 1000);
+    }
+    return () => timer && clearInterval(countdown);
+  }, [isRunning, timeRemaining, timer]);
+
+  const toggleTimer = () => {
+    setIsRunning(!isRunning);
+  };
+
+  const displayTime = (time) => {
+    const hours = Math.floor(time / 3600);
+    const minutes = Math.floor((time % 3600) / 60);
+    const seconds = time % 60;
+
+    return `${hours}:${minutes < 10 ? '0' : ''}${minutes}:${
```

Comment:
hours also needs HH format. e.g. `05` instead of just `5`. may create a function to return `x < 10 ? '0' : ''` since it will be used thrice.

Reply:
Fixed